src/util/evalUnits.ts

```diff
@@ -53,19 +52,37 @@ export enum EvalMessages {
  * the first is the actual calculated number (or null if none) while
  * the second is the message that should appear in case of error
  */
-export function evalUnits(input: string): [number | null, string] {
+export function evalUnits(input: string, chainDecimals: number): [BN | null, string] {
+  //sanitize input to remove + char if exists
+  input = input && input.replace('+', '');
   if (!floats.test(input) && !ints.test(input) && !alphaInts.test(input) && !alphaFloats.test(input)) {
     return [null, EvalMessages.GIBBERISH];
   }
+  // find the character from the alphanumerics
+  const symbol = input.replace(/[0-9.,]/g, '');
+  // find the value from the si list
+  const siVal = si.find((s) => s.symbol === symbol);
+  const numberStr = input.replace(symbol, '').replace(',', '.');
+  let numeric: BN = new BN(0);
+  if (siVal) {
+    let action: number | BN;
+    const containDecimal = input.replace(/[,]/g, '.').includes('.');
+    if (containDecimal) {
+      action = parseFloat(numberStr.replace(/[,]/g, '.')) * 10 ** chainDecimals;
```

Comment:
The point of my proposal (and comparing parsed float with the input) was to avoid parsing user inputs incorrectly. Your approach can be still:
1. silently lossy (`parseFloat('100000000000000000.1') == 100000000000000000`)
2. or can end up failing to be converted to `BN` (try `1000` with `18` decimals: `(1000.0 * 10**18).toString ==  '1e+21'`).